app/src/main/java/com/example/parkingsystem/mvp/view/ParkingReservationView.kt

```diff
@@ -54,50 +54,82 @@ class ParkingReservationView(activity: Activity, private val binding: ActivityPa
 
     override fun showMissingDateTimeStart() {
         context?.let {
-            it.showToast(it.getString(R.string.toast_parking_reservation_day_time_start_missing))
+            var dialog = ParkingCustomAlertDialog(it)
+            val title = it.getString(R.string.title_validation_warning)
+            val message = it.getString(R.string.text_parking_reservation_day_time_start_missing)
+            dialog.showAlertDialog(it, title, message)
+        }
+    }
+
+    override fun showExpiredDateTimeStart() {
+        context?.let {
+            var dialog = ParkingCustomAlertDialog(it)
+            val title = it.getString(R.string.title_validation_warning)
+            val message = it.getString(R.string.text_parking_reservation_day_time_start_expired)
+            dialog.showAlertDialog(it, title, message)
         }
     }
 
     override fun showMissingDateTimeEnd() {
         context?.let {
-            it.showToast(it.getString(R.string.toast_parking_reservation_day_time_end_missing))
+            var dialog = ParkingCustomAlertDialog(it)
+            val title = it.getString(R.string.title_validation_warning)
+            val message = it.getString(R.string.text_parking_reservation_day_time_end_missing)
+            dialog.showAlertDialog(it, title, message)
         }
     }
 
     override fun showInvalidDateTimeEnd() {
         context?.let {
-            it.showToast(it.getString(R.string.toast_parking_reservation_day_time_end_invalid))
+            var dialog = ParkingCustomAlertDialog(it)
+            val title = it.getString(R.string.title_validation_warning)
+            val message = it.getString(R.string.text_parking_reservation_day_time_end_invalid)
+            dialog.showAlertDialog(it, title, message)
         }
     }
 
     override fun showMissingParkingSpace() {
         context?.let {
-            it.showToast(it.getString(R.string.toast_parking_reservation_space_missing))
+            var dialog = ParkingCustomAlertDialog(it)
+            val title = it.getString(R.string.title_validation_warning)
+            val message = it.getString(R.string.text_parking_reservation_space_missing)
+            dialog.showAlertDialog(it, title, message)
         }
     }
 
     override fun showOutOfBoundParkingSpace(parkingLot: Int) {
         context?.let {
-            it.showToast(it.getString(R.string.toast_parking_reservation_space_out_of_bound,parkingLot))
+            var dialog = ParkingCustomAlertDialog(it)
```

Comment:
could be val this too

Reply:
It can be better, I can erase all of theme (because they are single use variables XD ). Working on that. Thanks a lot